src/main/duke/command/DeadlineCommand.java

```diff
@@ -0,0 +1,58 @@
+package duke.command;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.exception.DukeArgumentException;
+import duke.exception.DukeExecutionException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.ui.Ui;
+
+
+/**
+ * Represents an request by the user to add a Deadline object to the list of Tasks.
+ */
+public class DeadlineCommand extends Command {
+
+    private Deadline deadline;
+
+    /**
+     * Constructor for a DeadlineCommand object.
+     *
+     * @param input the input string to be parsed into the arguments for constructing a Deadline object.
+     * @throws DukeArgumentException if the arguments from the input string are invalid.
+     */
+    public DeadlineCommand(String input) throws DukeArgumentException {
+        String[] args = input.split("/by ", 2);
+        try {
+            this.deadline = new Deadline(args[0].trim(), LocalDate.parse(args[1].trim()));
+        } catch (ArrayIndexOutOfBoundsException ae) {
+            throw new DukeArgumentException("Insufficient arguments provided for duke.Tasks.Deadline.");
+        } catch (DateTimeParseException de) {
+            throw new DukeArgumentException("duke.Tasks.Deadline date/time could not be parsed.");
+        }
+    }
+
+    @Override
+    public boolean shouldExit() {

```

Comment:
Any reason why you don't choose to omit the shouldExit() method? 
[Same for other command classes that return false as well]
From my understanding, if the sub-class behaves exactly as the parent class, we can choose to omit it and the system can compile.

Reply:
Yes, you're right that it can be omitted. Will be removing this in the next commit, thanks.